hrui/hrui.php

```diff
@@ -224,12 +224,23 @@ function hrui_civicrm_install() {
   //delete default tag of civicrm
   CRM_Core_DAO::executeQuery("DELETE FROM civicrm_tag WHERE name IN ('Non-profit', 'Company', 'Government Entity', 'Major Donor', 'Volunteer')");
 
-  // make sure only relevant components are enabled
-  $params = array(
+  // make sure only relevant core components are enabled
+  $defaultCoreComponents = Civi::settings()->getDefault('enable_components');
+  $unwantedCoreComponents = array_diff($defaultCoreComponents, ['CiviReport','CiviCase']);
+
+  $getResult = civicrm_api3('setting', 'getsingle', [
     'domain_id' => CRM_Core_Config::domainID(),
-    'enable_components' => array('CiviReport','CiviCase'),
-  );
-  $result = civicrm_api3('setting', 'create', $params);
+    'return' => ['enable_components'],
+  ]);
+
+  $enabledComponents = $getResult['enable_components'];
+  $enableComponents = array_diff($enabledComponents, $unwantedCoreComponents);
```

Comment:
I don't understand why `hrui` is enabling/disabling components, as this extension is only supposed to have civicrm core UI customizations (e.g, hiding fields, labels, etc).

Another confusing thing is that it seems we have code that handles enabling components in at least three different parts of the system (hrui, [hrcase](https://github.com/compucorp/civihr/blob/513f9b86955132a8ba03e670ae1e61797ce4bcb4/hrcase/CRM/HRCase/Upgrader.php#L18-L20) and [t&a](https://github.com/compucorp/civihr-tasks-assignments/blob/staging/uk.co.compucorp.civicrm.tasksassignments/CRM/Tasksassignments/Upgrader.php#L20-L23)), which I think is the biggest reason we had all this confusion in the first place.

What I think we should do is:
- HRCore disables all the components
- T&A enables CiviCase, CiviTask and CiviDocument (We need to make sure CiviCase is not used by any other extension that gets installed before T&A)
- Check which extension actually uses CiviReport (`hrreport`?) and enable it there.

So, basically, hrcore will be responsible for only disabling components (which is different from hrui that is currently responsible for enabling and disabling them), and extensions are responsible for only enabling the ones they need (without disabling previously enabled ones).

Another option would be to move all the enabling/disabling logic to hrcore so that we have this in one single place. My only concern about this is that the CiviTask and CiviDocument componets are defined by T&A and I don't know if it would be possible to enabled these components before T&A is installed.